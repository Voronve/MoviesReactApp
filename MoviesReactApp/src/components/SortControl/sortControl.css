.sortBywrapper {
    min-width: 250px;
    font: 12px 'Montserrat', sans-serif;
}

.sortBywrapper nav {
    display: flex;
    justify-content: space-between;
    position: relative;
    height: 100%;
    align-items: center;
}

.sortBywrapper nav div {
    vertical-align: middle;
}

.dropdown-menu {
    font-size: 16px;
    color: white;
    margin: 0;
    cursor: pointer;
    position: absolute;
    top: 26%;
    right: 30px;
    z-index: 1;
}

.sortByText, .dropdown-menu li{
    color: #8b8888;
}

.dropdown-menu li {
    cursor: pointer;
    list-style: none;
    display: none;
}

.dropdown-menu li.active {
    display: block;
}

.dropdown-menu li:hover, .dropdown-menu li.active {
    color: white;
}

.sortBywrapper nav {
    position: relative;
}

.sortBywrapper:hover .dropdown-menu {
    display: flex;
    flex-direction: column;
}

.sortBywrapper:hover .dropdown-menu li{
    display: block;
}

.arrowDown {
    color: red;
}