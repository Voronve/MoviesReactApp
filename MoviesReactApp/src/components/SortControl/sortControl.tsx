import { useState } from "react";
import './sortControl.css';

export enum sortValue {
    default = "default",
    title = "title",
    date = "date"
}

interface sortControlProps {
    /** callback function to be called after sort option been clicked*/
    sortBy: (sortValue: sortValue) => void
}

/** Element for sorting movies by title or date*/
export function SortControl({ sortBy }: sortControlProps) {
    const [state, setState] = useState({activeTitleClass: "active", activeDateClass: ""});

    const handleClick = (value: sortValue) => {
        if(value == sortValue.title) {
            setState({...state, activeTitleClass: "active", activeDateClass: ""});
            sortBy(sortValue.title)
        } else {
            setState({...state, activeTitleClass: "", activeDateClass: "active"});
            sortBy(sortValue.date)
        }
      };

    return (
        <div className="sortBywrapper">
            <nav>
                <div className="sortByText">SORT BY</div>
                <ul className="dropdown-menu">
                    <li className={state.activeTitleClass} onClick={() => handleClick(sortValue.title)} >TITLE</li>
                    <li className={state.activeDateClass} onClick={() => handleClick(sortValue.date)} >RELEASE DATE</li>
                </ul>
                <div className="arrowDown">â–¼</div>
            </nav>
        </div>);
}