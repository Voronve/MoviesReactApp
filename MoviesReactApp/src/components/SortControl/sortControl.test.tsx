import {SortControl, sortValue} from './sortControl';
import { fireEvent, render, screen } from '@testing-library/react';

describe("Testing SortControl component", () => {
    const sortBy = jest.fn();

    test("Testing SortControl rendering", () => {

        const { asFragment } = render(<SortControl activeSorting={sortValue.default} sortBy={sortBy}/>);

        expect(asFragment()).toMatchSnapshot();
    });

    test("Testing that callback function is working as expected", () => {
        render(<SortControl activeSorting={sortValue.default} sortBy={sortBy}/>);

        fireEvent.click(screen.getByText("RELEASE DATE"));
        expect(sortBy).toBeCalledWith(sortValue.date);
        fireEvent.click(screen.getByText("TITLE"));
        expect(sortBy).toBeCalledWith(sortValue.title);
    });
});