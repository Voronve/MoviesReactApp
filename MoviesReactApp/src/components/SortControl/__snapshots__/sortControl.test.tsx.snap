// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SortControl component Testing SortControl rendering 1`] = `
<DocumentFragment>
  <div
    class="sortBywrapper"
  >
    <nav>
      <div
        class="sortByText"
      >
        SORT BY
      </div>
      <ul
        class="dropdown-menu"
      >
        <li
          class="active"
        >
          TITLE
        </li>
        <li
          class=""
        >
          RELEASE DATE
        </li>
      </ul>
      <div
        class="arrowDown"
      >
        â–¼
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
